vochain/app.go

```diff
@@ -257,11 +268,23 @@ func (app *BaseApplication) InitChain(_ context.Context,
 // CheckTx unmarshals req.Tx and checks its validity
 func (app *BaseApplication) CheckTx(_ context.Context,
 	req *abcitypes.RequestCheckTx) (*abcitypes.ResponseCheckTx, error) {
+	txReference := vochaintx.TxKey(req.Tx)
+	// store the initial height of the tx
+	initialTTLheight, isReferenced := app.txTTLReferences.LoadOrStore(txReference, app.Height())
+	// check if the tx is referenced by a previous block and the TTL has expired
+	if isReferenced && app.Height() > initialTTLheight.(uint32)+transactionBlocksTTL {
+		// remove tx reference and return checkTx error
+		log.Debugw("pruning expired tx from mempool", "height", app.Height(), "hash", fmt.Sprintf("%x", txReference))
+		app.txTTLReferences.Delete(txReference)
+		return &abcitypes.ResponseCheckTx{Code: 1, Data: []byte(fmt.Sprintf("tx expired %x", txReference))}, nil
+	}
```

Comment:
also, these lines are not covered by tests: https://coveralls.io/builds/62999336/source?filename=vochain%2Fapp.go#L276

it would be great if you could add a test or extend one of the existing ones. 

AFAIU, pushing a tx with a very high nonce would do it, right?
and then wait until it's pruned, then check somehow the result

i can also self-assign this little task as well if you prefer and push a commit to PR

Reply:
please, feel free to add the missing test :)